### ISSUE TITLE
cupy RuntimeError on triangle creation

### ISSUE THREAD

**Issue by cbalona** [casact/chainladder-python#89]
When creating a triangle from a long format dataframe, a runtime error is encountered.

```python
import chainladder as cl
cl.array_backend('cupy')
import pandas as pd

data = pd.read_csv('schedp/medmal_pos.csv')
triangle = cl.Triangle(data, columns=['IncurLoss_F2'], origin=['AccidentYear'], development=['DevelopmentYear'], cumulative=True)
````
```
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-5-046b9c101c59> in <module>
      1 data = pd.read_csv('schedp/medmal_pos.csv')
----> 2 triangle = cl.Triangle(data, columns=['IncurLoss_F2'], origin=['AccidentYear'], development=['DevelopmentYear'], cumulative=True)

c:\users\work\envs\mlonact\lib\site-packages\chainladder\core\base.py in __init__(self, data, origin, development, columns, index, origin_format, development_format, cumulative, array_backend, *args, **kwargs)
    130                 warnings.warn('Unable to load CuPY.  Using numpy instead.')
    131                 self.array_backend = 'numpy'
--> 132             self.values = xp.array(values, dtype=kwargs.get('dtype', None))
    133         self.is_cumulative = cumulative
    134 

c:\users\work\envs\mlonact\lib\site-packages\cupy\creation\from_data.py in array(obj, dtype, copy, order, subok, ndmin)
     41 
     42     """
---> 43     return core.array(obj, dtype, copy, order, subok, ndmin)
     44 
     45 

cupy\core\core.pyx in cupy.core.core.array()

cupy\core\core.pyx in cupy.core.core.array()

cupy\core\core.pyx in cupy.core.core._send_object_to_gpu()

c:\users\work\envs\mlonact\lib\site-packages\sparse\_sparse_array.py in __array__(self, *args, **kwargs)
    220 
    221         if not AUTO_DENSIFY:
--> 222             raise RuntimeError(
    223                 "Cannot convert a sparse array to dense automatically. "
    224                 "To manually densify, use the todense method."

RuntimeError: Cannot convert a sparse array to dense automatically. To manually densify, use the todense method.
```

[medmal_pos.zip](https://github.com/casact/chainladder-python/files/4987446/medmal_pos.zip)

**Comment by jbogaardt** [comment]
Thanks for reporting @cbalona. I seem to have let the maintenance of the the cupy backend go a little. Part of the problem is there is no free automated build service (that I am aware of) that will give me GPU access.  I'll have to grab a GPU enabled machine and run the tests manually.  I'm assuming this doesn't happen when you use the numpy backend?

**Comment by cbalona** [comment]
Hi, @jbogaardt. Works perfectly fine on the numpy backend. Using chainladder 0.7.4.

**Comment by jbogaardt** [comment]
cupy backend now working.  Will be released as part of chainladder 0.7.5
